server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -192,6 +197,72 @@ public Weight createWeight(Query query, ScoreMode scoreMode, float boost) throws
         }
     }
 
+    @Override
+    protected LeafSlice[] slices(List<LeafReaderContext> leaves) {
+        return computeSlices(leaves, queueSizeBasedExecutor.threadPoolExecutor.getPoolSize(), MINIMUM_DOCS_PER_SLICE);
+    }
+
+    /**
+     * Each computed slice contains at least 10% of the total data in the leaves with a
+     * minimum given by the <code>minDocsPerSlice</code> parameter and the final number
+     * of {@link LeafSlice} will be equal or lower than the number of available threads.
+     */
+    /* pkg private for testing */
+    static LeafSlice[] computeSlices(List<LeafReaderContext> leaves, int numThreads, int minDocsPerSlice) {
```

Comment:
is numThreads to be interpreted as max number of slices created? If so I'd rename the argument for clarity.

Reply:
make sense, I have renamed it.

Reply:
thank you